### PR TITLE
Move `ipFamily` config file field to `KubernetesNetworkConfig`

### PR DESCRIPTION
**Pull request by nikimanoledaki**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4564

This PR moves the config file field `ipFamily` from `vpc` to `kubernetesNetworkConfig`. 

Since `kubernetesNetworkConfig` is not guaranteed to exist, there was a bit of validation needed to ensure that `kubernetesNetworkConfig` isn't nil to avoid nil pointer dereference panics.

This PR also splits the validations for when the `ipFamily` field is set in general (e.g. addons, k8s version) and the validations for when `ipFamily` is set in combination with `vpc`-related params. 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -164,56 +164,23 @@ func (c *ClusterConfig) ValidateVPCConfig() error {
...
-		return fmt.Errorf("invalid value %s for ipFamily; allowed are %s and %s", c.VPC.IPFamily, IPV4Family, IPV6Family)
-	}
 
-	// This is the new vpc check, I need this check when the user sets it.
-	if c.VPC.IPFamily == IPV6Family {
-		if err := c.ipv6CidrsValid(); err != nil {
-			return err
+	if c.VPC.IPv6Cidr != "" || c.VPC.IPv6Pool != "" {
+		if c.KubernetesNetworkConfig.IPFamily != IPV6Family {
+			return fmt.Errorf("Ipv6Cidr and Ipv6CidrPool is only supportd when IPFamily is set to IPv6")
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:170]:
```suggestion
			return fmt.Errorf("Ipv6Cidr and Ipv6CidrPool are only supported when IPFamily is set to IPv6")
```

**Review by cPu1 — COMMENTED** [review]
I have proposed a minor suggestion but otherwise LGTM.

**Review by cPu1 — APPROVED** [review]
LGTM! :tada:

**Review by Himangini — APPROVED** [review]
🎉

**Pull request merged**